### PR TITLE
Require Node.js 8, add support for nano and Emacs, add TypeScript definition

### PR DESCRIPTION
**Pull request by BendingBender**
~~Build broken due to https://github.com/benmosher/eslint-plugin-import/issues/1322.~~

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+Get info about a specific editor.
+
+@param editor - This can be pretty flexible. It matches against all the data it has.
+
+For example, to get Sublime Text, you could write either of the following: `sublime`, `Sublime Text`, `subl`.
+@returns Info about the specified editor.
+
+@example
+```
+import * as envEditor from 'env-editor';
```
Comment (sindresorhus) [inline index.d.ts]:
```suggestion
import envEditor = require('env-editor');
```

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+@returns Info about the default editor.
+*/
+export default function(): Editor;
+
+/**
+Get info about a specific editor.
+
+@param editor - This can be pretty flexible. It matches against all the data it has.
+
```
Comment (sindresorhus) [inline index.d.ts]: It looks weird to have an empty line here, even though the above belongs to the `editor` description.

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+For example, to get Sublime Text, you could write either of the following: `sublime`, `Sublime Text`, `subl`.
+@returns Info about the specified editor.
```
Comment (sindresorhus) [inline index.d.ts]: `Info` => `Metadata`?

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
+export interface Editor {
+	id: string;
+	name: string;
+	bin: string;
```
Comment (sindresorhus) [inline index.d.ts]: I think we should rename `bin` to `binary`.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
I know you don't review the API thoroughly, but like I said before, if you see anything that could be improved in the API or otherwise, I'm all ears. Many of these modules I made years ago, and are not up to my current standards, documentation and API-wise.

**Comment by sindresorhus** [comment]
Thanks for adding additional editors ❤️

File: index.d.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+	isTerminalEditor: boolean;
+	paths: string[];
+	keywords: string[];
```
Comment (sindresorhus) [inline index.d.ts:7]: Same as other PR question, should these be `readonly`?

File: index.d.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+// 	isTerminalEditor: false,
+// 	paths: [
+// 		'/Applications/Sublime Text.app/Contents/SharedSupport/bin/subl',
+// 		'/Applications/Sublime Text 2.app/Contents/SharedSupport/bin/subl'
+// 	],
+// 	keywords: []
+// }
+```
+*/
+export function get(editor: string): Editor;
```
Comment (sindresorhus) [inline index.d.ts]: I don't think these should be named exports. Importing `{get}` is meaningless without context. I only do named methods when the imported named method name is descriptive enough individually.

**Comment by BendingBender** [comment]
I generally would avoid using `default` as a function name, it has this ambiguity with ES exports. What about renaming it to `defaultEditor`? Should we then rename the other two to `getEditor` and `allEditors`? What about named exports after renaming?

**Comment by sindresorhus** [comment]
> What about renaming it to defaultEditor? Should we then rename the other two to getEditor and allEditors? What about named exports after renaming?

Yeah, we can rename all the methods and make them named exports.

**Pull request merged**